src/sentry/models/outbox.py

```diff
@@ -0,0 +1,380 @@
+from __future__ import annotations
+
+import abc
+import contextlib
+import datetime
+import sys
+from enum import IntEnum
+from typing import Any, Generator, Iterable, List, Mapping, Set, TypeVar
+
+from django.db import connections, models, router, transaction
+from django.db.models import Max
+from django.dispatch import Signal
+from django.utils import timezone
+
+from sentry.db.models import (
+    BoundedBigIntegerField,
+    BoundedPositiveIntegerField,
+    JSONField,
+    Model,
+    control_silo_only_model,
+    region_silo_only_model,
+    sane_repr,
+)
+from sentry.silo import SiloMode
+
+THE_PAST = datetime.datetime(2016, 8, 1, 0, 0, 0, 0, tzinfo=timezone.utc)
```

Comment:
Any arbitrary fixed point of time in the past for which we can default the first queueing of items into the outbox queue.  By being a fixed timepoints it makes testing easier.